pkg/controllers/managementuser/nodesyncer/nodessyncer.go

```diff
@@ -851,3 +836,18 @@ func (m *nodesSyncer) isClusterRestoring() (bool, error) {
 	}
 	return false, nil
 }
+
+func determineNodeRoles(machine *v3.Node) {
+	if machine.Status.NodeLabels != nil {
+		_, etcd := machine.Status.NodeLabels["node-role.kubernetes.io/etcd"]
+		_, control := machine.Status.NodeLabels["node-role.kubernetes.io/controlplane"]
+		_, master := machine.Status.NodeLabels["node-role.kubernetes.io/master"]
+		_, worker := machine.Status.NodeLabels["node-role.kubernetes.io/worker"]
+		machine.Spec.Etcd = etcd
+		machine.Spec.Worker = worker
+		machine.Spec.ControlPlane = control || master
+		if !machine.Spec.Worker && !machine.Spec.ControlPlane && !machine.Spec.Etcd {
+			machine.Spec.Worker = true
+		}
+	}
+}
```

Comment:
Is this method split out just so its more testable ? I don't see any real changes

Reply:
yeah, same steps - just added tests